src/main/java/sg/ncl/webssh/WebSocketController.java

```diff
@@ -0,0 +1,150 @@
+package sg.ncl.webssh;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.messaging.handler.annotation.MessageMapping;
+import org.springframework.messaging.handler.annotation.Payload;
+import org.springframework.messaging.simp.SimpMessageHeaderAccessor;
+import org.springframework.stereotype.Controller;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * This controller is for a web socket to ssh into a node. The library used is JSch (Java Secure Channel).
+ *
+ * References:
+ * [1] http://kimrudolph.de/blog/spring-4-websockets-tutorial
+ * [2] http://www.sergialmar.com/2014/03/detect-websocket-connects-and-disconnects-in-spring-4/
+ * [3] https://docs.spring.io/spring-session/docs/current/reference/html5/guides/websocket.html
+ * [4] https://stackoverflow.com/questions/28387157/multiple-rooms-in-spring-using-stomp
+ * [5] https://stackoverflow.com/questions/29899720/how-to-execute-interactive-commands-and-read-their-sparse-output-from-java-using
+ * [6] https://stackoverflow.com/questions/41286714/get-websocket-session-data-in-spring-when-handing-stomp-send
+ * [7] https://github.com/skavanagh/KeyBox-OpenShift/blob/master/src/main/java/com/keybox/manage/socket/SecureShellWS.java
+ */
+
+@Controller
+@Slf4j
+public class WebSocketController {
+
+    private WebSocketBean webSocketBean;
+
+    @Autowired
+    public WebSocketController(WebSocketBean webSocketBean) {
+        this.webSocketBean = webSocketBean;
+    }
+
+    @MessageMapping("/input")
+    public void handleInput(WebSocketInput input, SimpMessageHeaderAccessor headerAccessor) {
+        Boolean ctrlKey = input.getCtrlKey();
+        String key = input.getKey();
+        Integer keyCode = input.getKeyCode();
+        if (!ctrlKey && ((keyCode >= 65 && keyCode <= 90) || keyCode == 219 || keyCode == 221)) {
+            keyCode = 0;
+        }
+        if (keyMap.containsKey(keyCode)) {
+            try {
+                webSocketBean.getInputToShell().write(keyMap.get(keyCode));
+            } catch (IOException ioe) {
+                ioe.printStackTrace();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use a logger to log this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1148)